apps/docs/pages/guides/database/postgres/configuration.mdx

```diff
@@ -12,14 +12,47 @@ Postgres provides a set of sensible defaults for you database size. In some case
 
 ## Timeouts
 
-By default, Supabase limits the maximum statement execution time to _8 seconds_ for users accessing the API. Additionally, all users are subject to a global limit of _2 minutes_. This serves as a backstop against resource exhaustion due to either poorly written queries, or abusive usage. This can be overridden using [Custom Postgres configuration](https://supabase.com/docs/guides/platform/custom-postgres-config#supported-parameters).
+By default, Supabase limits the maximum statement execution time to _8 seconds_ for users accessing the API. All role timeouts can be viewed with the following query:
+
+```sql
+SELECT
+  rolname, rolconfig
+FROM pg_roles
+WHERE rolname IN (
+  'anon',
+  'authenticated',
+  'authenticator',
+  'postgres',
+  'service_role'
+);
+```
+
+Additionally, all users are subject to a global limit of _2 minutes_. This serves as a backstop against resource exhaustion due to either poorly written queries, or abusive usage. To view the global timeout, you can execute the below query:
+
+```sql
+SHOW statement_timeout;
+```
+
+This can be overridden using [Custom Postgres configuration](https://supabase.com/docs/guides/platform/custom-postgres-config#supported-parameters). The below example is for changing the global timeout with `npx`:
+
+```bash
+#Logs you into the Supabase CLI
+npx supabase login
+
+#Change database timeout to 3 minutes
+npx supabase --experimental --project-ref <YOUR PROJECT REF> postgres-config update --config statement_timeout='3min'
+```
 
 ### Changing the default timeout
 
 The timeout values were picked as a reasonable default for the majority of use-cases, but can be modified using the [`alter role`](https://www.postgresql.org/docs/current/sql-alterrole.html) statement:
 
+<Admonition type="note" label="Self hosting">
+When updating either the `anon` or `authenticated` role, set the `authenticator` role to the greater timeout value between the two.
+</Admonition>
+
 ```sql
--- For API users
+alter role authenticated set statement_timeout = '15s';
 alter role authenticator set statement_timeout = '15s';
```

Comment:
Small nit but because of the precedence, I would put authenticator first as it takes precedence.

Also maybe keep a comment but put lines making it explicit that you would need to set them e.g

```
-- For API users, you need to set authenticator and then the authenticated value: 
alter role authenticator set statement_timeout = '16s';
alter role authenticated set statement_timeout = '15s';
```